compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -71,7 +71,8 @@ object ProtoTypes {
                    |constraint was: ${ctx.typerState.constraint}
                    |constraint now: ${newctx.typerState.constraint}""")
             if result && (ctx.typerState.constraint ne newctx.typerState.constraint) then
-              newctx.typerState.commit()
+              newctx.typerState.gc()     // Remove any type lambdas and tvars added via testCompat
+              newctx.typerState.commit() // Commit the rest of the typer state information
```

Comment:
It's not clear to me that this makes a difference: `commit()` will call `gc()` on the target state after moving type lambdas to that state, so I'd expect anything that `newctx.typerState.gc()` does to be subsumed by that?

Reply:
You were right again 😄.
So I slowed down and tried again.